sdk/core/System.ClientModel/src/Internal/BitVector640.cs

```diff
@@ -28,136 +28,66 @@ internal struct BitVector640
     private ulong _bits8;
     private ulong _bits9;
 
-    public bool this[int i]
+    // To get or set the correct bit, we need to find a bracket and an offset inside bracket
+    // For that, we have GetBracket and GetOffset methods.
+    // ===================================================
+    // Example: find the bracket and the offset for index 621
+    // In binary form, 621 is 0b1001101101.
+    // First 4 bits represent the bracket number, while the last 6 bits represent the offset in the bracket
+    // (ulong is 64-bit type, hence offset is in the [0, 63] range)
+    //
+    //    9 - bracket number
+    // ╔══╩══╗
+    // 1 0 0 1 1 0 1 1 0 1
+    //         ╚════╦════╝
+    //              45 - offset in the bracket
+    // ╔════════════╩════════════════════════════════════════════════════════╗
+    // 00000000 00000000 00100000 00000000 00000000 00000000 00000000 00000000
```

Comment:
I like the picture, but the words are confusing.

Reply:
Renamed `bracket` into `field`. Other suggestions are welcome.